stdlib/ostruct.rb

```diff
@@ -60,10 +66,52 @@ def to_n
   def hash
     @table.hash
   end
+  
+  def table
+    @table
+  end
+  
+  def delete_field(name)
+    sym = name.to_sym
+    begin
+      singleton_class.__send__(:remove_method, sym, "#{sym}=")
+    rescue NameError
+    end
+    @table.delete sym
+  end
+  
+  def new_ostruct_member(name)
```

Comment:
That makes sense. That said, the rubyspecs have several tests for each of those (that expect to send to those methods since they are protected/private). It would be a shame to filter/lose those specs, which I think we would have to do if we turned them into local helpers (I assume you mean a lambda or something inside the class definition when you say local helper).